The rule manager should store the User. r=Yoric

ferjm opened the pull request:
This adds the owner of the rule to the scripts DB and also send it to the corresponding adapter when a rule is executed.

Yoric commented:
In the future, could you add "r? @Yoric" in the body? I have a filter that detects these to make sure I don't miss any review request.

Yoric commented on src/manager.rs:
```diff
@@ -66,7 +66,8 @@ impl<Env, T> ScriptManager<Env, T>
             source      TEXT NOT NULL,
-            is_enabled  BOOL NOT NULL DEFAULT 1
+            is_enabled  BOOL NOT NULL DEFAULT 1,
+            owner       INTEGER NOT NULL DEFAULT -1
```
Can you add documentation? All other fields of the database are basically self-documenting, but that one really isn't.

Yoric commented on src/manager.rs:
```diff
@@ -104,12 +110,17 @@ impl<Env, T> ScriptManager<Env, T>
 
     /// Attempt to add a new script. The script will be executed and persisted to disk.
     /// The ID is chosen by the consumer and must be unique.
```
Documentation needed.

ferjm commented:
> In the future, could you add "r? @Yoric" in the body? I have a filter that detects these to make sure I don't miss any review request.

I added and removed it cause I needed to rebase :)

Yoric commented on src/manager.rs:
```diff
@@ -192,14 +204,27 @@ impl<Env, T> ScriptManager<Env, T>
+    /// Get the owner of a script given its id.
+    pub fn get_owner(&self, id: &Id<ScriptId>) -> Result<User, Error> {
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        let mut stmt = try!(connection.prepare("SELECT owner FROM scripts WHERE id = $1"));
```
I don't think that you need a `prepare` here.

Yoric commented on src/manager.rs:
```diff
@@ -135,7 +146,8 @@ impl<Env, T> ScriptManager<Env, T>
             },
             (true, false) => {
-                try!(self.start_script(id, &source));
+                let owner = try!(self.get_owner(&id));
```
Mmmh... so we are opening and closing the connection twice for a single change? That looks wasteful/fragile.

Yoric commented on tests/test_manager.rs:
```diff
@@ -33,7 +34,8 @@ fn test_database_add_remove_script() {
 
     let name = Id::<ScriptId>::new("Sample Ruleset");
-    db.put(&name, &load_json("./examples/ruleset.json")).unwrap();
+    db.put(&name, &load_json("./examples/ruleset.json"), &User::Id(1)).unwrap();
```
Ideally, we would like a test that ensures that the user id sent to the destination channel is correct. I realize that this is more complex, but can you 

1/ Test it manually to confirm that it works;
2/ File a followup bug
?

ferjm replied:
https://github.com/fxbox/thinkerbell/issues/71

Yoric commented:
r=me with a few changes
